cognite/client/_api_client.py

```diff
@@ -986,12 +986,12 @@ def _upsert_multiple(
         if mode not in ["patch", "replace"]:
             raise ValueError(f"mode must be either 'patch' or 'replace', got {mode!r}")
         is_single = isinstance(items, CogniteResource)
-        items = cast(Sequence[CogniteResource], [items] if is_single else items)
+        items = cast(Sequence[T_CogniteResource], [items] if is_single else items)
         try:
             result = self._update_multiple(items, list_cls, resource_cls, update_cls, mode=mode)
         except CogniteNotFoundError as not_found_error:
-            items_by_external_id = {item.external_id: item for item in items if item.external_id is not None}
-            items_by_id = {item.id: item for item in items if hasattr(item, "id") and item.id is not None}
+            items_by_external_id = {item.external_id: item for item in items if item.external_id is not None}  # type: ignore [attr-defined]
```

Comment:
Shouldn't this also have a `hasattr(item, "external_id")`. I think there are resources that do not have an `external_id`.

Reply:
I wondered the same thing, but actaully wanted to ask you this - as I think you wrote this code `(_upsert_multiple)`?